### PR TITLE
Except validate_form_action from selective_domain

### PR DESCRIPTION
**Pull request by Tawakalt**
**Proposed changes**:
- Fixes [ATO-785](https://rasahq.atlassian.net/browse/ATO-785)

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

[ATO-785]: https://rasahq.atlassian.net/browse/ATO-785?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docs/domain.mdx [hunk 1]
```diff
@@ -809,13 +809,15 @@ action_endpoint:
 ```
 **After selective domain for custom actions is enabled, domain will be sent only to
 those custom actions which have specifically stated that they need it.**
+`validate_form_action` is an exception to this rule as it will always have the domain sent to it.
```
Comment (radovanZRasa) [inline docs/docs/domain.mdx:812]: This is gonna be true for any action which starts with `validate_`.
@ancalita 
Do you maybe know it we have other actions which start with `validate_`?

Comment (ancalita) [reply docs/docs/domain.mdx:812]: @Tawakalt That is correct, I think we need to remove the code part (e.g. `validate_form_action`) and instead say something more generic like `custom actions inheriting from rasa-sdk FormValidationAction parent class (link to the part of documentations for this)`

Comment (Tawakalt) [reply docs/docs/domain.mdx:812]: That makes perfect sense @ancalita

File: docs/docs/domain.mdx [hunk 2]
```diff
@@ -809,13 +809,16 @@ action_endpoint:
 ```
 **After selective domain for custom actions is enabled, domain will be sent only to
 those custom actions which have specifically stated that they need it.**
+Custom actions inheriting from rasa-sdk [FormValidationAction](https://rasa.com/docs/rasa/action-server/validation-action/#formvalidationaction-class)
```
Comment (ancalita) [inline docs/docs/domain.mdx:812]: you can use dynamic reference instead:
```suggestion
Custom actions inheriting from rasa-sdk [FormValidationAction](./action-server/validation-action.mdx#formvalidationaction-class)
```

**Review by ancalita — APPROVED** [review]
LGTM 💯

**Pull request merged**